source/common/router/router.cc

```diff
@@ -652,9 +652,6 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
   }
   callbacks_->streamInfo().setAttemptCount(attempt_count_);
 
-  // Inject the active span's tracing context into the request headers.
-  callbacks_->activeSpan().injectContext(headers);
```

Comment:
What are the implications of this with regard to retries and hedging? I'm concerned this a change in behavior here?

Reply:
If `start_child_span` is set to `false`, then we may do some repeated injectings when retries occur. This should be safe because  in previous implementation, we may re-inject tracing context as well.

Reply:
In the #20367, we want to update the `injectContext` API to add a new parameter to provide target upstream of current upstream request. This PR can also complete some pre-work.